var/spack/repos/builtin/packages/rocm-openmp-extras/package.py

```diff
@@ -0,0 +1,337 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+import os
+import re
+
+tools_url = 'https://github.com/ROCm-Developer-Tools'
+compute_url = 'https://github.com/RadeonOpenCompute'
+
+# Arrays of hashes are in order of the versions array below
+# For example array[0] = 3.9.0, array[1] = 3.10.0, etc.
+
+aomp = [
+    "377ab59b685a73b3f95fba95f5e028678ec5aafabc4177b7f0ffb78da095d679",
+    "808fca9bdefb109d5bcbbc9f5b59c564a6d422488869e986516f2a7233eda235"
+]
+
+devlib = [
+    "c99f45dacf5967aef9a31e3731011b9c142446d4a12bac69774998976f2576d7",
+    "bca9291385d6bdc91a8b39a46f0fd816157d38abb1725ff5222e6a0daa0834cc"
+]
+
+llvm = [
+    "1ff14b56d10c2c44d36c3c412b190d3d8cd1bb12cfc7cd58af004c16fd9987d1",
+    "8262aff88c1ff6c4deb4da5a4f8cda1bf90668950e2b911f93f73edaee53b370"
+]
+
+flang = [
+    "5d113f44fb173bd0d5704b282c5cebbb2aa642c7c29f188764bfa1daa58374c9",
+    "3990d39ff1c908b150f464f0653a123d94be30802f9cad6af18fbb560c4b412e"
+]
+
+extras = [
+    "830a37cf1c6700f81fc00749206a37e7cda4d2867bbdf489e9e2d81f52d06b3d",
+    "5d98d34aff97416d8b5b9e16e7cf474580f8de8a73bd0e549c4440a3c5df4ef5"
+]
+
+versions = ['3.9.0', '3.10.0']
+versions_dict = dict()
+components = ['aomp', 'devlib', 'llvm', 'flang', 'extras']
+component_hashes = [aomp, devlib, llvm, flang, extras]
+
+# Loop through versions and create necessary dictionaries of components
+for outer_index, item in enumerate(versions):
+    for inner_index, component in enumerate(component_hashes):
+        versions_dict.setdefault(item, {})[components[inner_index]] = \
+            component_hashes[inner_index][outer_index]
+
+
+class RocmOpenmpExtras(Package):
+    """OpenMP support for ROCm LLVM."""
+
+    homepage = tools_url + "/aomp"
+    url = tools_url + "/aomp/archive/rocm-uc-3.10.0.tar.gz"
+
+    maintainers = ['srekolam', 'arjun-raj-kuppala', 'estewart08']
+    version('3.10.0', sha256=versions_dict['3.10.0']['aomp'])
+    version('3.9.0', sha256=versions_dict['3.9.0']['aomp'])
+
+    depends_on('cmake@3.5.2:3.13.4', type='build')
```

Comment:
It won't build with newer CMake?

Reply:
It very well could. Is it expected of a recipe to test every possible version combination of its dependencies? If so, would this range suffice for now?

Reply:
It's not expected. If you don't know, I wouldn't add an upper limit. The lower limit generally comes from the `CMakeLists.txt`.

Reply:
In my opinion, removing the upper limit has the potential of breaking the recipe for someone.  I would rather have a recipe that takes a bit longer to build over a non-functional one, but I can definitely see the speed benefit if the newer version does work.